hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/command/ws/WsCommands.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.cmdgw.command.ws;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.hawkular.bus.common.BasicMessage;
+import org.hawkular.cmdgw.api.EchoRequest;
+import org.hawkular.cmdgw.api.ResourcePathDestination;
+import org.hawkular.cmdgw.api.UiSessionDestination;
+import org.hawkular.cmdgw.command.NoCommandForMessageException;
+
+/**
+ * A registry of {@link WsCommand} that can operate on messages coming over a WebSocket.
+ *
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+@ApplicationScoped
+public class WsCommands {
+
+    private final ResourcePathDestinationWsCommand resourcePathDestinationWsCommand = //
+    new ResourcePathDestinationWsCommand();
+
+    private final EchoCommand echoCommand = new EchoCommand();
+
+    private final UiSessionDestinationWsCommand uiSessionDestinationWsCommand = new UiSessionDestinationWsCommand();
+
+    /**
+     * Returns a {@link WsCommand} that should handle the given {@code requestClass}.
+     *
+     * @param requestClass the type of a request for which a processing {@link WsCommand} should be found by this
+     *        method
+     * @return a {@link WsCommand}, never {@code null}
+     * @throws NoCommandForMessageException if no {@link WsCommand} was found
+     */
+    @SuppressWarnings("unchecked")
+    public <REQ extends BasicMessage> WsCommand<REQ> getCommand(Class<REQ> requestClass)
+            throws NoCommandForMessageException {
+        if (ResourcePathDestination.class.isAssignableFrom(requestClass)) {
```

Comment:
put a "//TODO" or some comment to indicate this is a place developers will need to manually add code if new command or request types are added in the future.